Fix Type Annotations in pandas/_config/config.py

Part of #25882

Current errors are as follows:

```sh
pandas/_config/config.py:60: error: Need type annotation for '_deprecated_options'
pandas/_config/config.py:61: error: Need type annotation for '_registered_options'
pandas/_config/config.py:62: error: Need type annotation for '_global_config'
```

Should add annotations and remove this module from mypy.ini - think this is straightforward so community PR is certainly welcome!

--- a/mypy.ini
+++ b/mypy.ini
@@ -5,9 +5,6 @@ follow_imports=silent
 [mypy-pandas.conftest,pandas.tests.*]
 ignore_errors=True
 
-[mypy-pandas._config.config]
-ignore_errors=True
-
 [mypy-pandas._version]
 ignore_errors=True
 

--- a/pandas/_config/config.py
+++ b/pandas/_config/config.py
@@ -58,8 +58,11 @@ DeprecatedOption = namedtuple('DeprecatedOption', 'key msg rkey removal_ver')
 RegisteredOption = namedtuple('RegisteredOption',
                               'key defval doc validator cb')
 
+# holds deprecated option metdata
 _deprecated_options = {}  # type: Dict[str, DeprecatedOption]
+# holds registered option metdata
 _registered_options = {}  # type: Dict[str, RegisteredOption]
+# holds the current values for registered options
 _global_config = {}  # type: Dict[str, str]
 _reserved_keys = ['all']  # keys which have a special meaning
 

--- a/pandas/_config/config.py
+++ b/pandas/_config/config.py
@@ -51,15 +51,16 @@ Implementation
 from collections import namedtuple
 from contextlib import contextmanager
 import re
+from typing import Dict
 import warnings
 
 DeprecatedOption = namedtuple('DeprecatedOption', 'key msg rkey removal_ver')
 RegisteredOption = namedtuple('RegisteredOption',
                               'key defval doc validator cb')
 
-_deprecated_options = {}  # holds deprecated option metdata
-_registered_options = {}  # holds registered option metdata
-_global_config = {}  # holds the current values for registered options
+_deprecated_options = {}  # type: Dict[str, str]
+_registered_options = {}  # type: Dict[str, object]
+_global_config = {}  # type: Dict[str, str]
 _reserved_keys = ['all']  # keys which have a special meaning
 
 

--- a/pandas/_config/config.py
+++ b/pandas/_config/config.py
@@ -58,8 +58,8 @@ DeprecatedOption = namedtuple('DeprecatedOption', 'key msg rkey removal_ver')
 RegisteredOption = namedtuple('RegisteredOption',
                               'key defval doc validator cb')
 
-_deprecated_options = {}  # type: Dict[str, str]
-_registered_options = {}  # type: Dict[str, object]
+_deprecated_options = {}  # type: Dict[str, DeprecatedOption]
+_registered_options = {}  # type: Dict[str, RegisteredOption]
 _global_config = {}  # type: Dict[str, str]
 _reserved_keys = ['all']  # keys which have a special meaning
 

--- a/pandas/_config/config.py
+++ b/pandas/_config/config.py
@@ -51,7 +51,7 @@ Implementation
 from collections import namedtuple
 from contextlib import contextmanager
 import re
-from typing import Dict
+from typing import Dict, List
 import warnings
 
 DeprecatedOption = namedtuple('DeprecatedOption', 'key msg rkey removal_ver')
@@ -60,11 +60,15 @@ RegisteredOption = namedtuple('RegisteredOption',
 
 # holds deprecated option metdata
 _deprecated_options = {}  # type: Dict[str, DeprecatedOption]
+
 # holds registered option metdata
 _registered_options = {}  # type: Dict[str, RegisteredOption]
+
 # holds the current values for registered options
 _global_config = {}  # type: Dict[str, str]
-_reserved_keys = ['all']  # keys which have a special meaning
+
+# keys which have a special meaning
+_reserved_keys = ['all']  # type: List[str]
 
 
 class OptionError(AttributeError, KeyError):

